test/parallel/test-http-connect.js

```diff
@@ -1,85 +1,90 @@
 'use strict';
 const common = require('../common');
-var assert = require('assert');
-var http = require('http');
+const assert = require('assert');
+const http = require('http');
 
-var serverGotConnect = false;
-var clientGotConnect = false;
+let serverGotConnect = false;
+let clientGotConnect = false;
 
-var server = http.createServer(common.fail);
-server.on('connect', function(req, socket, firstBodyChunk) {
-  assert.equal(req.method, 'CONNECT');
-  assert.equal(req.url, 'google.com:443');
+const server = http.createServer(common.fail);
+
+server.on('connect', common.mustCall((req, socket, firstBodyChunk) => {
+  assert.strictEqual(req.method, 'CONNECT');
+  assert.strictEqual(req.url, 'google.com:443');
   console.error('Server got CONNECT request');
   serverGotConnect = true;
 
   socket.write('HTTP/1.1 200 Connection established\r\n\r\n');
 
-  var data = firstBodyChunk.toString();
-  socket.on('data', function(buf) {
+  let data = firstBodyChunk.toString();
+  socket.on('data', common.mustCall((buf) => {
```

Comment:
I would avoid adding `common.mustCall()` to the `data` handler. The number of events emitted can potentially be different, and it isn't really needed for the test.